[2024-10-10T15:50:43.671+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pipeline_A.task_0_S3_to_DB manual__2024-10-10T15:42:50.567022+00:00 [queued]>
[2024-10-10T15:50:43.676+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pipeline_A.task_0_S3_to_DB manual__2024-10-10T15:42:50.567022+00:00 [queued]>
[2024-10-10T15:50:43.676+0000] {taskinstance.py:1288} INFO - 
--------------------------------------------------------------------------------
[2024-10-10T15:50:43.676+0000] {taskinstance.py:1289} INFO - Starting attempt 2 of 2
[2024-10-10T15:50:43.676+0000] {taskinstance.py:1290} INFO - 
--------------------------------------------------------------------------------
[2024-10-10T15:50:43.684+0000] {taskinstance.py:1309} INFO - Executing <Task(PythonOperator): task_0_S3_to_DB> on 2024-10-10 15:42:50.567022+00:00
[2024-10-10T15:50:43.702+0000] {standard_task_runner.py:55} INFO - Started process 31999 to run task
[2024-10-10T15:50:43.708+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'pipeline_A', 'task_0_S3_to_DB', 'manual__2024-10-10T15:42:50.567022+00:00', '--job-id', '16', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_A.py', '--cfg-path', '/tmp/tmpge63r37w']
[2024-10-10T15:50:43.711+0000] {standard_task_runner.py:83} INFO - Job 16: Subtask task_0_S3_to_DB
[2024-10-10T15:50:43.727+0000] {logging_mixin.py:137} WARNING - /usr/local/lib/python3.9/site-packages/***/settings.py:250 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-10-10T15:50:43.803+0000] {task_command.py:389} INFO - Running <TaskInstance: pipeline_A.task_0_S3_to_DB manual__2024-10-10T15:42:50.567022+00:00 [running]> on host 1f7125be5d21
[2024-10-10T15:50:43.859+0000] {taskinstance.py:1516} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=pipeline_A
AIRFLOW_CTX_TASK_ID=task_0_S3_to_DB
AIRFLOW_CTX_EXECUTION_DATE=2024-10-10T15:42:50.567022+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2024-10-10T15:42:50.567022+00:00
[2024-10-10T15:50:43.862+0000] {s3_db.py:108} INFO - Executing S3 File read.......
[2024-10-10T15:50:43.862+0000] {s3_db.py:18} INFO - Creating Spark session ......
[2024-10-10T15:50:50.080+0000] {s3_db.py:44} INFO - Executing File pattern check......
[2024-10-10T15:50:50.786+0000] {taskinstance.py:1776} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/airflow/operators/python.py", line 175, in execute
    return_value = self.execute_callable()
  File "/usr/local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/scripts/s3_db.py", line 282, in main_s3_db
    inputs = s3_file_read(s3_conn, source_s3_path, header, delimiter)
  File "/opt/airflow/scripts/s3_db.py", line 140, in s3_file_read
    file_name = file_pattern_check(source_s3_path, s3_conn)
  File "/opt/airflow/scripts/s3_db.py", line 94, in file_pattern_check
    latest_file = max(match_list, key=extract_timestamp)
  File "/usr/local/lib/python3.9/site-packages/pyspark/sql/utils.py", line 160, in wrapped
    return f(*args, **kwargs)
TypeError: max() got an unexpected keyword argument 'key'
[2024-10-10T15:50:50.802+0000] {taskinstance.py:1327} INFO - Marking task as FAILED. dag_id=pipeline_A, task_id=task_0_S3_to_DB, execution_date=20241010T154250, start_date=20241010T155043, end_date=20241010T155050
[2024-10-10T15:50:50.823+0000] {standard_task_runner.py:100} ERROR - Failed to execute job 16 for task task_0_S3_to_DB (max() got an unexpected keyword argument 'key'; 31999)
[2024-10-10T15:50:50.865+0000] {local_task_job.py:212} INFO - Task exited with return code 1
[2024-10-10T15:50:50.917+0000] {taskinstance.py:2596} INFO - 0 downstream tasks scheduled from follow-on schedule check
